### PR TITLE
Exclude virtual_ymca_editor from category access check 2

### PR DESCRIPTION
**Pull request by anpolimus**
**Related Issue/Ticket:**

https://github.com/ymcatwincities/openy_gated_content/issues/77

## Steps to test:

- [ ] log in  and create a user with the "Virtual YMCA Editor" role
- [ ] observe that as Administrator and Member, you can see these pages:
  - [ ] /virtual-ymca#/categories/blog
  - [ ] /virtual-ymca#/categories/video
- [ ] log in as the Editor
- [ ] Observe the above pages are empty, with no error or message.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [x] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA team in Slack

### CODE REVIEW

File: src/VirtualYAccessTrait.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * @package Drupal\openy_gated_content
+ */
+trait VirtualYAccessTrait {
+
+  /**
+   * Constant with virtual y editor role.
+   *
+   * @var string
+   */
+  public static $virtualYAccessEditorRole = 'virtual_ymca_editor';
```
Comment (froboy) [inline src/VirtualYAccessTrait.php:17]: @anpolimus this still only allows a single `virtualYAccessEditorRole`. Would it be possible to make this an array which devs could extend/override? Some of the other check logic would have to be changed, but i think it's only 2 or 3 places.

Comment (anpolimus) [reply src/VirtualYAccessTrait.php:17]: we can make it not a variable but a function

**Review by hamrant — APPROVED** [review]

**Comment by anpolimus** [comment]
retest

File: src/VirtualYAccessTrait.php (hunk 1 above)
Comment (anpolimus) [reply src/VirtualYAccessTrait.php:17]: @froboy, i've changed approach from variable to a method that returns an array

Comment (froboy) [reply src/VirtualYAccessTrait.php:17]: Nice. I think that will work. I'm not able to test immediately, but visual review looks good.

**Review by hamrant — APPROVED** [review]

**Comment by NadyaRudenko** [comment]
tested, works as expected. thanks!

**Pull request merged**